src/Engines/Nunjucks.js

```diff
@@ -121,7 +122,35 @@ export default class Nunjucks extends TemplateEngine {
 
 	addFilters(filters, isAsync) {
 		for (let name in filters) {
-			this.njkEnv.addFilter(name, Nunjucks.wrapFilter(name, filters[name]), isAsync);
+			let callback = filters[name];
+			if (!isAsync) {
+				/** @this {any} */
+				callback = function (...args) {
+					// Note that `callback` is already a function as the `#add` method throws an error if not.
+					let ret = filters[name].call(this, ...args);
+					if (ret instanceof Promise) {
+						throw new Error(
```

Comment:
Where is this supposed to be catched?

Reply:
It’s not. This is the same code addFilter added to nunjucks. It has been relocated to the nunjucks file for the purpose of decoupling